datastore/datastore.go

```diff
@@ -14,9 +13,9 @@ import (
 //DataStore exported
 type DataStore interface {
 	// GetObject gets data from datastore and unmarshals to the specified object
-	GetObject(key string, o interface{}) error
+	GetObject(key string, o KV) error
 	// GetUpdatedObject gets data from datastore along with its index and unmarshals to the specified object
-	GetUpdatedObject(key string, o interface{}) (uint64, error)
+	GetUpdatedObject(key string, o KV) (uint64, error)
```

Comment:
Thanks for the change !

Now that you fixed GetObject() signature, can we have this function set the db index  before returning the object to the caller and get rid of GetUpdatedObject() function.

I added GetUpdatedObject() in a rush just to get around the fact GetObject() did not provide the set index.